File: common/src/main/java/org/syncloud/common/jackson/RFC822DateTimeDeserializer.java
package org.syncloud.common.jackson;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;

import java.io.IOException;
import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class RFC822DateTimeDeserializer extends JsonDeserializer<Date> {
    @Override
    public Date deserialize(JsonParser jsonparser, DeserializationContext deserializationcontext) throws IOException, JsonProcessingException {
        DateFormat format = new SimpleDateFormat();
        String dateStr = jsonparser.getText();
        Date date = null;
        try {
            date = format.parse(dateStr);
        } catch (ParseException e) {
            throw new IOException("Can't parse DateTime from RFC-822 format", e);
        }
        return date;
    }
}

File: syncloud/src/main/java/org/syncloud/android/core/common/jackson/RFC822DateTimeDeserializer.kt
package org.syncloud.android.core.common.jackson

import com.fasterxml.jackson.databind.JsonDeserializer
import kotlin.Throws
import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import java.io.IOException
import java.text.DateFormat
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class RFC822DateTimeDeserializer : JsonDeserializer<Date?>() {
    @Throws(IOException::class, JsonProcessingException::class)
    override fun deserialize(
        jsonparser: JsonParser,
        deserializationcontext: DeserializationContext
    ): Date? {
        val format: DateFormat = SimpleDateFormat()
        val dateStr = jsonparser.text
        var date: Date? = null
        date = try {
            format.parse(dateStr)
        } catch (e: ParseException) {
            throw IOException("Can't parse DateTime from RFC-822 format", e)
        }
        return date
    }
}